### PR TITLE
Convert CMakeSettings.json to CMakePresets.json for coreclr

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Part of #88688.

Basically, add all variables from cmake_cmd_line.txt.

The presets file is currently only used by IDE, but conversion of command line build for CLR should be straightforward. Currently only supports Windows because I didn't test other platforms.
CMakePresets supports file importing and we can separated the file when fully depending on it.

Corehost is a bit complicated and I'll try to include it later.

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]
Contributes to https://github.com/dotnet/runtime/issues/88688

LGTM!

File: src/coreclr/CMakePresets.json [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    "minor": 20
+  },
+  "configurePresets": [
+    {
+      "name": "base",
+      "hidden": true,
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/../../artifacts/obj/coreclr/${presetName}",
+      "cacheVariables": {
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/../../artifacts/bin/coreclr/${presetName}"
```
Comment (huoyaoyuan) [inline src/coreclr/CMakePresets.json:14]: The `installDir` property is available since version 3.

File: src/coreclr/CMakePresets.json [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      }
+    },
+    {
+      "name": "windows-base",
+      "hidden": true,
+      "inherits": "base",
+      "cacheVariables": {
+        "CLR_CMAKE_TARGET_OS": "windows",
+        "CLI_CMAKE_FALLBACK_OS": "win10"
```
Comment (huoyaoyuan) [inline src/coreclr/CMakePresets.json:23]: The varaible CMAKE_SYSTEM_VERSION is not set here. It seems to match exact Windows SDK version.

File: src/coreclr/CMakePresets.json [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      },
+      "cacheVariables": {
+        "CLR_CMAKE_HOST_ARCH": "x64"
+      }
+    },
+    {
+      "name": "x86",
+      "hidden": true,
+      "architecture": {
+        "value": "x86",
```
Comment (huoyaoyuan) [inline src/coreclr/CMakePresets.json:69]: MSBuild uses Win32 while Ninja uses x86 here. May be difficult to implement a `-msbuild` switch in the future.

File: src/coreclr/CMakePresets.json [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  "version": 2,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 20
+  },
+  "configurePresets": [
+    {
+      "name": "base",
+      "hidden": true,
+      "generator": "Ninja",
```
Comment (huoyaoyuan) [inline src/coreclr/CMakePresets.json:11]: Default since version 3.

**Review by huoyaoyuan — COMMENTED** [review]
Some notes for future potentials.

**Comment by huoyaoyuan** [comment]
Test failure looks unrelated. The files aren't affecting CLI build.
After getting merged, utilizing CMakeUserPresets.json for experiments would be easier.

**Pull request merged**